### PR TITLE
Enable stake pool relays in cli stake pool registration certificate creation

### LINKED ISSUE #1282: [FR] CLI - Support pool relays in pool registration-certificate command
https://github.com/input-output-hk/cardano-node/blob/478230484677e1940291eff6004e93d71692a242/cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs#L1314

### PR DESCRIPTION
**Pull request by Jimbo4350**
Closes #1282

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 1]
```diff
@@ -1251,24 +1251,76 @@ pPoolMargin =
...
+
+pSingleHostName :: Parser ShelleyStakePoolRelay
+pSingleHostName =
+  Shelley.SingleHostName <$> (Shelley.maybeToStrictMaybe <$> optional pPort) <*> pDNSName
+ where
+  pDNSName :: Parser Shelley.DnsName
+  pDNSName = Opt.option (Opt.eitherReader eDNSName)
+               (  Opt.long "host-name-pool-relay-dns"
+               <> Opt.metavar "STRING"
+               <> Opt.help "The stake pool relay's DNS name (An A or AAAA DNS record)"
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1276]:
```suggestion
               (  Opt.long "single-host-pool-relay"
               <> Opt.metavar "STRING"
               <> Opt.help "The stake pool relay's DNS name that corresponds to an A or AAAA DNS record"
```

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 2]
```diff
@@ -1251,24 +1251,76 @@ pPoolMargin =
...
+
+pMultiHostName :: Parser ShelleyStakePoolRelay
+pMultiHostName =
+  Shelley.MultiHostName <$> (Shelley.maybeToStrictMaybe <$> optional pPort) <*> pDNSName
+ where
+  pDNSName :: Parser Shelley.DnsName
+  pDNSName = Opt.option (Opt.eitherReader eDNSName)
+               (  Opt.long "multi-host-name-pool-relay-dns"
+               <> Opt.metavar "STRING"
+               <> Opt.help "The stake pool relay's DNS name (An SRV DNS record)"
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1265]: Perhaps something more like:

```suggestion
               (  Opt.long "multihost-pool-relay"
               <> Opt.metavar "STRING"
               <> Opt.help "The stake pool relay's DNS name that corresponds to an SRV DNS record"
```

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 3]
```diff
@@ -109,7 +109,7 @@ pAddressCmd =
   Opt.subparser $
     mconcat
       [ Opt.command "key-gen"
-          (Opt.info pAddressKeyGen $ Opt.progDesc "Create a single address key pair.")
+          (Opt.info pAddressKeyGen $ Opt.progDesc "Create a  address key pair.")
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:112]: Strange, I didn't specifically make that change.

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 4]
```diff
@@ -1251,24 +1251,76 @@ pPoolMargin =
...
+pIpV6 = Opt.option (Opt.maybeReader readMaybe :: Opt.ReadM IP.IPv6)
+           (  Opt.long "pool-relay-ipv6"
+           <> Opt.metavar "STRING"
+           <> Opt.help "The stake pool relay's IpV6 address"
+           )
+
+pPort :: Parser Shelley.Port
+pPort = Opt.option (fromInteger <$> Opt.eitherReader readEither)
+           (  Opt.long "pool-relay-port"
+           <> Opt.metavar "INT"
```
[the remark this answers is not in the document]
Comment (erikd) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1321]: Nah, disagree. `short` is a C programming idea and many users would not know what it was. `int` is reasonably widely understood to mean `Integer`.

[the remark this answers is not in the document]
Comment (intricate) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1323]: That's fair. I suppose I was mostly concerned about accurately portraying the upper limit on the port value.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**